pyris/api/app.py

```diff
@@ -86,6 +86,37 @@ def get(self, code):
         return iris
 
 
+@api.route('/city/code/<string:code>')
+class IrisListFromCityCode(Resource):
+    @api.doc(parser=api.parser(),
+             description="Look for all the iris codes in the city matching a city code")
+    def get(self, code):
+        Logger.info("looking for IRIS in the city %s", code)
+        iris=extract.get_iris_list_by_city_code(code)
+        if not iris:
+            api.abort(404, "City code '{}' not found".format(code))
+        return iris
+
+
+@api.route("/city/search/<string:query>")
+class IrisListFromQuery(Resource):
+    @api.doc(parser=api.parser(),
+             description="Get the list of all the iris codes in the city matching the query")
+    def get(self, query):
+        Logger.info("Looking for the list of iris in the city matching the query %s", query)
+        Logger.info("Looking for longitude and latitude for the query %s", query)
+        coord=address.coordinate(query)
+        if coord["address"] is None:
+            # Requests sent to '/api/search' that match nothing do not return a 404 error like other requests.
+            # Is this an intentional choice?
```

Comment:
Is that a good question. I think we can add a `api.abort(404)` to the other requests.